dev/benchmarks/platform_channels_benchmarks/lib/main.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:microbenchmarks/common.dart';
+
+List<Object> _makeTestBuffer(int size) {
+  final List<Object> answer = <Object>[];
+  for (int i = 0; i < size; ++i) {
+    switch (i % 9) {
+      case 0:
+        answer.add(1);
+        break;
+      case 1:
+        answer.add(math.pow(2, 65));
+        break;
+      case 2:
+        answer.add(1234.0);
+        break;
+      case 3:
+        answer.add(null);
+        break;
+      case 4:
+        answer.add(<int>[1234]);
+        break;
+      case 5:
+        answer.add(<String, int>{'hello': 1234});
+        break;
+      case 6:
+        answer.add('this is a test');
+        break;
+      case 7:
+        answer.add(true);
+        break;
+      case 8:
+        answer.add(Uint8List(64));
+        break;
+    }
+  }
+  return answer;
+}
+
+Future<void> _runTests() async {
+  assert(false,
+      "Don't run benchmarks in checked mode! Use 'flutter run --release'.");
+  const int numMessages = 10000;
+
+  const BasicMessageChannel<Object> channel = BasicMessageChannel<Object>(
+      'dev.flutter.echo.basic.standard', StandardMessageCodec());
+  final Stopwatch watch = Stopwatch();
+
+  watch.start();
+  for (int i = 0; i < numMessages; ++i) {
+    await channel.send(1234);
+  }
+  watch.stop();
+  final double smallPayloadTime = watch.elapsedMicroseconds / numMessages;
+
+  watch.reset();
+  final List<Object> largeBuffer = _makeTestBuffer(1000);
+  const StandardMessageCodec codec = StandardMessageCodec();
+  final ByteData data = codec.encodeMessage(largeBuffer);
+  print('Large buffer size: ${data.lengthInBytes}');
+  int size = 0;
+  watch.start();
+  for (int i = 0; i < numMessages; ++i) {
+    final List<Object> result = await channel.send(largeBuffer) as List<Object>;
+    // This check should be tiny compared to the actual channel send/receive.
+    size += (result == null) ? 0 : result.length;
+  }
+  watch.stop();
+  final double largePayloadTime = watch.elapsedMicroseconds / numMessages;
+
+  if (size != largeBuffer.length * numMessages) {
+    throw Exception(
+        'There is an error with the echo channel, the results don\'t add up: $size');
+  }
+
+  final BenchmarkResultPrinter printer = BenchmarkResultPrinter();
+  printer.addResult(
+    description:
+        '  BasicMessageChannel/StandardMessageCodec/Flutter->Host/Small',
```

Comment:
Why the leading space in the string?

Reply:
I was following this example: https://github.com/flutter/flutter/blob/8fe4cc79648a952f9c7e49a5248756c2ff98fa3b/dev/benchmarks/microbenchmarks/lib/stocks/animation_bench.dart#L93

I'll remove it.  I thought it was weird at the time, too.

Reply:
done